enhancements/api-review/custom-tags-aws.md

```diff
@@ -0,0 +1,144 @@
+---
+title: apply-user-defined-tags-to-all-aws-resources-created-by-openshift
+authors:
+  - "@gregsheremeta"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-03-24
+last-updated: 2021-03-24
+status: implementable
+---
+
+# Apply user defined tags to all AWS resources created by OpenShift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes a proposal to allow an administrator of OpenShift to
+have the ability to apply user defined tags to all resources created by OpenShift in AWS.
+
+Note: this enhancement is slightly retroactive. Work has already begun on this. See
+ - https://github.com/openshift/api/pull/864
+ - https://github.com/openshift/cluster-ingress-operator/pull/578
+
+## Motivation
+
+Motivations include but are not limited to:
+
+ - allow admin, compliance, and security teams to keep track of assets and objects created by OpenShift,
+   both at install time during continuous operation (Day 2)
+ - in a Managed OpenShift environment such as Red Hat OpenShift on AWS (ROSA), allow easy differentiation
+   between Red Hat-managed AWS objects and customer-managed objects
+ - *allow for the restriction of permissions granted to Red Hat in an AWS account by AWS resource tags.
+   For example, see SDE-1146 - "IAM users and roles can only operate on resources with specific tags"*
+
+### Goals
+
+ - the administrator or service (in the case of Managed OpenShift) installing OpenShift can pass an arbitrary
+   list of user-defined tags to the OpenShift Installer, and everything created by the installer and all other
+   bootstrapped components will apply those tags to all resources created in AWS, for the lift of the cluster.
+ - Installer writes the tags to the infrastructure resource so that Day 2 operations can know what tags to apply
+   to AWS resources for the life of the cluster.
+ - tags must be applied at creation time, in an atomic operation. It isn't acceptable to create an object and,
+   after some period of time, apply the tags post-creation.
+
+### Non-Goals
+
+ - to reduce initial scope, tags are applied only at creation time and not reconciled. If an administrator manually
+   changes the tags stores in the infrastructure resource, pre-existing AWS resources are not updated. Only newly-
+   created AWS resources would get the updated tags.
+ - to reduce initial scope, we're not implementing this for clouds other than AWS. We shouldn't take any actions to
+   prohibit that later, though.
+
+## Proposal
+
+Add a new field `userTags` to `.spec.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
```

Comment:
Is the user allowed to edit anything in this resource today?  Will there be any validation on the supplied values?

Reply:
Validation is something I want to understand too. I mentioned the tag restrictions here in the api PR: https://github.com/openshift/api/pull/864#issuecomment-805656042

Reply:
for this experimental pass, using status